### PR TITLE
Add and use kind-projector

### PR DESCRIPTION
**Pull request by dwijnand**

### CODE REVIEW

**Comment by ShaneDelmore** [comment]
That’s some pretty nice improvements to readability 👍

File: internal/util-collection/src/main/scala/sbt/internal/util/INode.scala [hunk 1]
```diff
@@ -32,27 +32,25 @@ abstract class EvaluateSettings[Scope] {
...
-      case o: Optional[s, T] @unchecked =>
-        o.a match {
-          case None    => constant(() => o.f(None))
-          case Some(i) => single[s, T](transform(i), x => o.f(Some(x)))
-        }
-      case x if x == StaticScopes =>
-        strictConstant(allScopes.asInstanceOf[T]) // can't convince scalac that StaticScopes => T == Set[Scope]
-    }
+  private[this] val transform: Initialize ~> INode = λ[Initialize ~> INode] {
+    case k: Keyed[s, A1$] @unchecked => single(getStatic(k.scopedKey), k.transform)
```
Comment (eed3si9n) [inline internal/util-collection/src/main/scala/sbt/internal/util/INode.scala:36]: What's this dollar sign mean here?

**Review by eed3si9n — CHANGES_REQUESTED** [review]
A minor comment.

File: internal/util-collection/src/main/scala/sbt/internal/util/INode.scala (hunk 1 above)
Comment (dwijnand) [reply internal/util-collection/src/main/scala/sbt/internal/util/INode.scala:36]: `A1$` is the name of the generated type parameter of `~>`'s apply, which was `T` before.

Basically this is what I found works that allows use to use polymorphic lambda syntax _even_ in places where we need the type param. It's more on the risky side of things in this diff. Btw there are 2 cases in InputTask.scala where I couldn't find any way to use the syntax, so it uses old school anonymous inner classes.

File: internal/util-collection/src/main/scala/sbt/internal/util/Attributes.scala [hunk 2]
```diff
@@ -196,11 +196,9 @@ object AttributeMap {
   def apply(entries: AttributeEntry[_]*): AttributeMap = empty ++ entries
 
   /** Presents an `AttributeMap` as a natural transformation. */
-  implicit def toNatTrans(map: AttributeMap): AttributeKey ~> Id = new (AttributeKey ~> Id) {
-    def apply[T](key: AttributeKey[T]): T = map(key)
-  }
-
+  implicit def toNatTrans(map: AttributeMap): AttributeKey ~> Id = λ[AttributeKey ~> Id](map(_))
```
Comment (eed3si9n) [inline internal/util-collection/src/main/scala/sbt/internal/util/Attributes.scala:199]: I am perfectly happy to go with uber fp polyfunction _value level_ λ notation, but I am really not sure if this increases the "readability."

**Review by eed3si9n — APPROVED** [review]

File: internal/util-collection/src/main/scala/sbt/internal/util/Attributes.scala (hunk 2 above)
Comment (dwijnand) [reply internal/util-collection/src/main/scala/sbt/internal/util/Attributes.scala:199]: Yeah.. I'm happy to see the reduction in boilerplate, but I'm still not 100% convinced.

**Pull request merged**